src/ai/defaultai.cc

```diff
@@ -3398,24 +3462,31 @@ bool DefaultAI::construct_building(const Time& gametime) {
 				if (bf->is_portspace != ExtendedBool::kTrue && bo.is(BuildingAttribute::kPort)) {
 					continue;
 				}
-				prio += bo.primary_priority;
 
-				// iterating over current warehouses and testing a distance
-				// getting distance to nearest warehouse and adding it to a score
-				uint16_t nearest_distance = std::numeric_limits<uint16_t>::max();
-				for (const WarehouseSiteObserver& wh_obs : warehousesites) {
-					const uint16_t actual_distance =
-					   map.calc_distance(bf->coords, wh_obs.site->get_position());
-					nearest_distance = std::min(nearest_distance, actual_distance);
-				}
-				// but limit to 30
-				const uint16_t max_distance_considered = 30;
-				nearest_distance = std::min(nearest_distance, max_distance_considered);
-				if (nearest_distance < 13) {
```

Comment:
I think we should keep this safeguard but we can use average_flag_dist_to_wh.
unrelated change might be to increase this a bit (lets say 20) as AI tends to build warehouses to close together

Reply:
This code is completely removed. This was checking air distance and is replaced completely by that flag distance

Reply:
I know and this is fine. However I still believe we should not allow to build any Warehouse if average_flag_dist_to_wh is lower then 15 to 20 for this field

Reply:
I was only talking about this if block not the rest

Reply:
OK, I added a condition at least 15